cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDeployClientConfigProvider.java

```diff
@@ -0,0 +1,261 @@
+            }
+        }
+        return result;
+    }
+
+    // similar as apiClient#Call creator methods - so keep this in one place
+    private Request createRequest(ApiClient apiClient, Map<String, List<String>> headers)
+            throws ApiException {
+        List<Pair> queryParams = new ArrayList<>();
+        Map<String, String> headerParams = new HashMap<>();
+        String[] localVarAccepts = new String[]{"application/json"};
+        String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);
+        if (localVarAccept != null) {
+            headerParams.put("Accept", localVarAccept);
+        }
+        String[] localVarContentTypes = new String[0];
+        String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);
+        headerParams.put("Content-Type", localVarContentType);
+        String[] authNames = new String[]{"basic"};
+        enhanceRequestParams(headers, queryParams, headerParams);
+        Request defaultRequest = apiClient.buildRequest(CMF_COMMANDS_COMMAND_TABLE_URI_PATH, "GET", queryParams, null,
+                headerParams, new HashMap<>(), authNames, null);
+        return createRequestFromDefaultRequest(defaultRequest);
+    }
+
+    private Request createRequestFromDefaultRequest(Request defaultRequest) {
+        HttpUrl defaultHttpUrl = defaultRequest.httpUrl();
+        HttpUrl.Builder httpUrlBuilder = new HttpUrl.Builder()
+                .username(defaultHttpUrl.username())
+                .password(defaultHttpUrl.password())
+                .host(defaultHttpUrl.host())
+                .scheme(defaultHttpUrl.scheme())
+                .port(defaultHttpUrl.port());
+        List<String> pathSegments = Splitter.on("/")
+                .splitToList(CMF_COMMANDS_COMMAND_TABLE_URI_PATH.substring(1));
+        for (String pathSegment : pathSegments) {
+            httpUrlBuilder.addPathSegment(pathSegment);
+        }
+        for (String queryParamName : defaultHttpUrl.queryParameterNames()) {
+            httpUrlBuilder.addQueryParameter(queryParamName, defaultHttpUrl.queryParameter(queryParamName));
+        }
+        return new Request.Builder()
+                .headers(defaultRequest.headers())
+                .url(httpUrlBuilder.build())
+                .cacheControl(defaultRequest.cacheControl())
+                .tag(defaultRequest.tag())
+                .get()
+                .build();
+    }
+
+    /**
+     * Extends request that is created based on the ApiClient endpoint implementations
+     * with default query params + Cookie header from listActiveCommands call.
+     * Example query: /cmf/commands/commandTable?limit=501&startTime=1234&endTime1235
+     */
+    private void enhanceRequestParams(Map<String, List<String>> headers, List<Pair> queryParams, Map<String, String> headerParams) {
+        queryParams.add(new Pair("limit", "501"));
+        long startTime = new DateTime()
+                .minusMinutes(INTERVAL_MINUTES)
```

Comment:
I feel 30 minutes is a bit much anyway.

Reply:
seems a lot but as commandTable is the 3rd api call, in a bad case it's possible the commands before that can take 8mins, maybe even the date on the CB host machine can be shifted, a 1 hour range for querying safer than like 30 minutes (note that commandTable does not contain every subcommand only the larger ones)